### PR TITLE
feature/BC-1464 - Implementing finish action for task cards

### PR DESCRIPTION
**Pull request by murat-merdoglu-hpi**
# Short Description
- Finishing and restoring features are added to the Task Cards.

## Links to Ticket and related Pull-Requests
- https://ticketsystem.dbildungscloud.de/browse/BC-1464
- https://github.com/hpi-schul-cloud/dof_app_deploy/pull/235

## Checklist before merging

- [x] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] PO: Any deviation from requirements was agreed with Product-Owner / ticket author / support-team

> Notice: Please keep this Pull-Request as a Draft (or add WIP label), until it is ready to be reviewed

### CODE REVIEW

File: src/components/templates/RoomDashboard.unit.ts [hunk 1]
```diff
@@ -405,4 +409,50 @@ describe("@components/templates/RoomDashboard.vue", () => {
 			expect(wrapper.vm.itemDelete.isOpen).toBe(false);
 		});
 	});
+
+	describe("Finishing and Restoring Tasks", () => {
+		it("should call finishTask action", async () => {
```
Comment (MartinSchuhmacher) [inline src/components/templates/RoomDashboard.unit.ts:414]: naming the role in the description might be better to distinguish the tests from each other

File: src/middleware/redirects.js [hunk 2]
```diff
@@ -1,21 +1,21 @@
...
+		// 	to: "/tasks",
+		// },
+		// {
+		// 	from: /\/homework\/private\/?/,
+		// 	to: "/tasks",
+		// },
+		// {
+		// 	from: /\/homework\/archive\/?/,
+		// 	to: "/tasks",
+		// },
```
Comment (MartinSchuhmacher) [inline src/middleware/redirects.js:18]: why are they commented out instead of deleted?

File: src/store/room.unit.ts [hunk 3]
```diff
@@ -570,6 +571,185 @@ describe("room module", () => {
...
+				(() => {
+					initializeAxios({
+						$patch: async (path: string, params: {}) => {
+							return { _id: "returnId" };
+						},
+						$get: async (path: string, params: {}) => {
+							received.push({ path });
+							received.push({ params });
+							return {
+								archived: ["asdasd"],
```
Comment (MartinSchuhmacher) [inline src/store/room.unit.ts:592]: maybe another test string would fit better here.

File: src/middleware/redirects.js (hunk 2 above)
Comment (murat-merdoglu-hpi) [reply src/middleware/redirects.js:18]: Because it will be considered in Nuxt Arch group.

**Review by MartinSchuhmacher — APPROVED** [review]

**Pull request merged**